pyproject.toml

```diff
@@ -102,3 +102,31 @@ ignore = [
 
 # Ignore too many leading '#' for block comment
 "*/tests/*" = ["E266"]
+
+
+[tool.pyright]
+pythonVersion = "3.10"
+# DevGoal: "strict"
+typeCheckingMode = "standard"
+include = [
+  "icepyx",
+]
+exclude = [
+  "**/__pycache__",
+  "icepyx/tests",
+]
+# DevGoal: Remove all ignores
+ignore = [
+  "icepyx/quest/*",
+  "icepyx/core/APIformatting.py",
```

Comment:
I want to remove this ignore first, but I didn't want this PR to get to huge. But I can also include that change if requested :)

Reply:
Does pyright ignore places there aren't type annotations? Without having run pyright yet, I'm trying to understand why these would currently fail (and need to be ignored).

Reply:
No, where there are no annotations, Pyright or Mypy will do inference. It's capable of finding bugs that way, too!

For example, `typing_test.py`:

```python
foo = "bar"
foo / 2
```

```bash
$ mypy typing_test.py 
typing_test.py:2: error: Unsupported operand types for / ("str" and "int")  [operator]
Found 1 error in 1 file (checked 1 source file)

$ pyright typing_test.py 
WARNING: there is a new pyright version available (v1.1.377 -> v1.1.378).
Please install the new version or set PYRIGHT_PYTHON_FORCE_VERSION to `latest`

/tmp/typing_test.py
  /tmp/typing_test.py:2:1 - error: Operator "/" not supported for types "Literal['bar']" and "Literal[2]" (reportOperatorIssue)
  /tmp/typing_test.py:2:1 - warning: Expression value is unused (reportUnusedExpression)
1 error, 1 warning, 0 informations 
```

Reply:
Got it - I ran it and now I'm seeing all the inferred issues it's catching.